src/cursor/abstract_cursor.ts

```diff
@@ -144,7 +145,13 @@ export abstract class AbstractCursor<
   /** @internal */
   [kNamespace]: MongoDBNamespace;
   /** @internal */
-  [kDocuments]: List<TSchema>;
+  [kDocuments]: {
+    length: number;
+    shift(bsonOptions?: any): TSchema | null;
```

Comment:
List.shift will get bson options passed into it that it will make no use of. We won't accidentally introduce new list API usage that isn't supported by CursorResponse